underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java

```diff
@@ -79,6 +83,9 @@ public boolean supportsPath(String path, UnderFileSystemConfiguration conf) {
       // are supported this is not an option and we have to continue to use this method.
       for (final String prefix : alluxioConf.getList(PropertyKey.UNDERFS_HDFS_PREFIXES, ",")) {
         if (path.startsWith(prefix)) {
+          if (conf != null && conf.get(PropertyKey.UNDERFS_VERSION).equals(getVersion())) {
```

Comment:
can we move the version-matching logic to HdfsVersion? Then it would be 
```java
if (conf == null || HdfsVersion.matches(conf.get(PropertyKey.UNDERFS_VERSION), getVersion())) {
  return true;
}
```

Reply:
I think it should be this below. Correct me if I'm wrong?

`if (conf != null && HdfsVersion.matches(conf.get(PropertyKey.UNDERFS_VERSION), getVersion())) {
  return true;
}`

Reply:
I'm thinking we can have `matches` check both string equality and equality after calling `HdfsVersion.find` on both strings. Then the code could be simplified to
```java
if (path.startsWith(prefix)) {
  if (conf == null || HdfsVersion.matches(conf.get(PropertyKey.UNDERFS_VERSION), getVersion())) {
    return true;
  }
}
```

or even 

```java
if (path.startsWith(prefix) &&
  (conf == null || HdfsVersion.matches(conf.get(PropertyKey.UNDERFS_VERSION), getVersion())) {
  return true;
}
```

Reply:
haha this is exactly what I suggest too @aaudiber  , essentially, @ns1123  this is the logic we'd better hide from `HdfsUnderFileSystemFactory.java` to `HdfsVersion`

Reply:
Moved the logic into HdfsVersion.java